autogluon/task/base/base_task.py

```diff
@@ -74,3 +82,70 @@ def run_fit(cls, train_fn, search_strategy, scheduler_options):
     @abstractmethod
     def fit(cls, *args, **kwargs):
         pass
+
+
+# These search_strategies use HyperbandScheduler, along with certain
+# searchers.
+searcher_for_hyperband_strategy = {
+    'hyperband': 'random'}
+
+
+def compile_scheduler_options(
+        search_strategy, nthreads_per_trial, ngpus_per_trial, checkpoint,
+        num_trials, time_out, resume, visualizer, time_attr, reward_attr,
+        search_options, dist_ip_addrs, **kwargs):
+    assert isinstance(search_strategy, str)
+    if visualizer is None:
+        visualizer = 'none'
+    if time_attr is None:
+        time_attr = 'epoch'
+    if reward_attr is None:
+        reward_attr = 'accuracy'
```

Comment:
Actually, I will follow your advice below and use a dict scheduler_options passed into run, instead of all these single ones. Then, defaults are dealt with by the scheduler itself.